client/state/current-user/gravatar-status/actions.js

```diff
@@ -20,20 +20,13 @@ import {
 	withAnalytics,
 } from 'state/analytics/actions';
 
-export function uploadGravatar( file, bearerToken, email ) {
+export function uploadGravatar( file, email ) {
 	return dispatch => {
 		dispatch( withAnalytics(
 			recordTracksEvent( 'calypso_edit_gravatar_upload_start' ),
 			{ type: GRAVATAR_UPLOAD_REQUEST }
 		) );
-
-		const data = new FormData();
-		data.append( 'filedata', file );
-		data.append( 'account', email );
-		return request
-			.post( 'https://api.gravatar.com/v1/upload-image' )
-			.send( data )
-			.set( 'Authorization', 'Bearer ' + bearerToken )
+		return wpcom.undocumented().uploadGravatar( email, file )
```

Comment:
I'd be happy to do so, but I've never used it. I suppose it depends on how quickly we want this to work. If there's no rush, it would be awesome to remove this thunk altogether and use the data layer to handle the network requests. On the other hand, since it has no effect on the user experience that could be done in a separate PR. I'll look into what using the new system would look like.